packages/table/src/table.tsx

```diff
@@ -897,6 +919,22 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         }
     }
 
+    private areChildrenSameShape(newChildArray: Array<React.ReactElement<IColumnProps>>) {
```

Comment:
I'd like unit tests for this. I'm not sure how much I can trust it to handle various cases at a glance.

Reply:
Got rid of this. Feels safer to only use this optimization if the children are shallow equal.